_includes/header.html

```diff
@@ -21,7 +21,7 @@
         <div class="trigger">
           {% for path in page_paths %}
             {% assign my_page = site.pages | where: "path", path | first %}
-            {% if my_page.title %}
+            {% if my_page.title and !my_page.exclude != true %}
```

Comment:
I'd recommend this be something more descriptive, like `exclude_from_nav` to prevent collisions with existing YAML.